### PR TITLE
fix(learn): change test order to test for the value after execution has finished

### LINKED ISSUE #39362: Avoid Mutations and Side Effects incomplete error testing
**Describe your problem and how to reproduce it:**

Part of the problem description states:

> Let's try to master this discipline and not alter any variable or object in our code

Therefore the following should not pass, however it does:

```
var fixedValue = 4;

function incrementer () {
  return ++fixedValue;
}
```

**Add a Link to the page with the problem:**
https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/avoid-mutations-and-side-effects-using-functional-programming.english.md

### PR DESCRIPTION
**Pull request by multiwebinc**
Checklist:

- [ x ] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [ x ] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [ x ] My pull request targets the `master` branch of freeCodeCamp.
- [ x ] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #39362

### CODE REVIEW

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/avoid-mutations-and-side-effects-using-functional-programming.english.md [hunk 1]
```diff
@@ -26,10 +26,10 @@ Fill in the code for the function <code>incrementer</code> so it returns the val
 
 ```yml
 tests:
-  - text: Your function <code>incrementer</code> should not change the value of <code>fixedValue</code> (which is <code>4</code>).
-    testString: assert(fixedValue === 4);
   - text: Your <code>incrementer</code> function should return a value that is one larger than the <code>fixedValue</code> value.
     testString: const newValue = incrementer(); assert(newValue === 5);
+  - text: Your function <code>incrementer</code> should not change the value of <code>fixedValue</code> (which is <code>4</code>).
+    testString: assert(fixedValue === 4);
```
Comment (nhcarrigan) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/avoid-mutations-and-side-effects-using-functional-programming.english.md:32]:
```suggestion
  - text: Your function <code>incrementer</code> should not change the value of <code>fixedValue</code> (which is <code>4</code>).
    testString: incrementer(); assert(fixedValue === 4);
```

Changing the order itself does not appear to be sufficient for testing - adding a function call before the assert statement will verify that the value of `fixedValue` has not changed.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

**Comment by RandellDawson** [comment]
@multiwebinc I like @nhcarrigan's suggestion, but also, I would leave the order of the tests the same, but also include @nhcarrigan's suggestion.

@nhcarrigan See the other issue I created for this challenge.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/avoid-mutations-and-side-effects-using-functional-programming.english.md (hunk 1 above)
Comment (nhcarrigan) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/avoid-mutations-and-side-effects-using-functional-programming.english.md:32]:
```suggestion
  - text: Your function <code>incrementer</code> should not change the value of <code>fixedValue</code> (which is <code>4</code>).
    testString: incrementer(); assert(fixedValue === 4);
  - text: Your <code>incrementer</code> function should return a value that is one larger than the <code>fixedValue</code> value.
    testString: const newValue = incrementer(); assert(newValue === 5);
```

This suggestion restores the original order of the tests, per @RandellDawson's comment.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

**Review by RandellDawson — APPROVED** [review]

**Review by naomi-lgbt — APPROVED** [review]
LGTM!

**Pull request merged**